### PR TITLE
Add update email views.

### PR DESCRIPTION
**Pull request by sylmak**
Add update email views.

Notes:
* Unlike the mocks, the update email form also asks for the password. This is because updating the email requires a recent login.
* The handle action path is based on this: https://firebase.google.com/docs/auth/custom-email-handler

### CODE REVIEW

File: src/components/UpdateEmail.vue [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  <h2>Update email</h2>
+  <form @submit.prevent>
+    <div class="current-email">
+      <label for="current-email">Current email</label>
+      <input type="text" name="current-email" :value="currentEmail" readonly />
+    </div>
+
+    <div class="new-email">
+      <label for="new-email">New email</label>
+      <input type="text" name="new-email" v-model="newEmail" required />
```
Comment (skeshive) [inline src/components/UpdateEmail.vue:52]: Missing the "We'll send you an email verification link" disclaimer here

File: src/components/UpdateEmailCheckEmail.vue [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+async function resendEmail() {
+  try {
+    await verifyBeforeUpdateEmail(auth.currentUser!, props.newEmail);
+  } catch (error) {
+    // TODO: Show error in UI.
+    console.log(error);
+    return;
+  }
+  alert('Update Email Sent!');
```
Comment (skeshive) [inline src/components/UpdateEmailCheckEmail.vue]: Did you mean to check this in?

File: src/router.ts [hunk 3]
```diff
@@ -18,6 +21,15 @@ export const router = createRouter({
     },
     // Account Management
     { path: '/account-management', component: AccountManagement },
+    // Handle Action
+    {
+      path: '/action',
+      component: HandleAction,
```
Comment (skeshive) [inline src/router.ts:28]: qq: why is this named generically? will we use the HandleAction component for other actions too?

File: src/components/HandleAction.vue [hunk 4]
```diff
@@ -0,0 +1,44 @@
+<script setup lang="ts">
```
Comment (skeshive) [inline src/components/HandleAction.vue:3]: can you add a quick comment for why this component is needed / what it does?

File: src/router.ts (hunk 3 above)
Comment (sylmak) [reply src/router.ts:28]: Yes, it will be used for other actions too, like reset password.

File: src/components/UpdateEmailCheckEmail.vue (hunk 2 above)
Comment (sylmak) [reply src/components/UpdateEmailCheckEmail.vue]: Yes, currently there's no proper indication in the UI that the email was resent.

Comment (skeshive) [reply src/components/UpdateEmailCheckEmail.vue]: Can we show it in the UI instead? Maybe under the resend button as a disclaimer

File: src/components/UpdateEmailCheckEmail.vue [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+</script>
+
+<template>
+  <h2>Check your email</h2>
+  <p>
+    We've sent an email to {{ props.newEmail }}. Please click on the link in
+    your email to verify.
+  </p>
+  <p>
+    Didn't receive the mail? Check your spam filter for an email from
```
Comment (skeshive) [inline src/components/UpdateEmailCheckEmail.vue]: email

**Review by skeshive — APPROVED** [review]

**Pull request merged**